packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -123,39 +91,33 @@ class Firestore {
       {Duration timeout = const Duration(seconds: 5)}) async {
     assert(timeout.inMilliseconds > 0,
         'Transaction timeout must be more than 0 milliseconds');
-    final int transactionId = _transactionHandlerId++;
-    _transactionHandlers[transactionId] = transactionHandler;
-    final Map<String, dynamic> result = await channel
-        .invokeMapMethod<String, dynamic>(
-            'Firestore#runTransaction', <String, dynamic>{
-      'app': app.name,
-      'transactionId': transactionId,
-      'transactionTimeout': timeout.inMilliseconds
-    });
+
+    // Wrap the transactionHandler into something that can be passed to the Platform implementation
+    final PlatformTransactionHandler handler = (int transactionId) async {
+      Transaction transaction = Transaction(transactionId, this);
+      final dynamic result = await transactionHandler(transaction);
+      await transaction._finish();
+      return result;
+    };
+
+    // Move to a runTransaction method in the method channel!
+    final Map<String, dynamic> result = await platform.runTransaction(app.name, transactionHandler: handler, transactionTimeout: timeout.inMilliseconds,);
     return result ?? <String, dynamic>{};
   }
 
   @deprecated
   Future<void> enablePersistence(bool enable) async {
     assert(enable != null);
-    await channel
-        .invokeMethod<void>('Firestore#enablePersistence', <String, dynamic>{
-      'app': app.name,
-      'enable': enable,
-    });
+    await platform.enablePersistence(app.name, enable: enable,);
   }
 
   Future<void> settings(
       {bool persistenceEnabled,
       String host,
       bool sslEnabled,
       int cacheSizeBytes}) async {
-    await channel.invokeMethod<void>('Firestore#settings', <String, dynamic>{
-      'app': app.name,
-      'persistenceEnabled': persistenceEnabled,
-      'host': host,
-      'sslEnabled': sslEnabled,
-      'cacheSizeBytes': cacheSizeBytes,
-    });
+    await platform.settings(app.name, persistenceEnabled: persistenceEnabled,
+        host: host, sslEnabled: sslEnabled,
+        cacheSizeBytes: cacheSizeBytes);
```

Comment:
readability nit: my preference would be a trailing comma and give each argument its own line here, assuming the formatter agrees.

Reply:
Absolutely, I was trying to convince dartfmt to do it automatically for me (that's why I left the trailing comma at the end), but it didn't take my hint. I'll format all of these manually :'(